components/engine_traits/src/options.rs

```diff
@@ -94,6 +100,26 @@ impl IterOptions {
         self.fill_cache = v;
     }
 
+    #[inline]
+    pub fn set_hint_min_ts(&mut self, ts: u64) {
+        self.hint_min_ts = Some(ts)
+    }
+
+    #[inline]
+    pub fn hint_min_ts(&self) -> Option<u64> {
+        self.hint_min_ts
+    }
+
+    #[inline]
+    pub fn hint_max_ts(&self) -> Option<u64> {
+        self.hint_max_ts
+    }
+
+    #[inline]
+    pub fn set_hint_max_ts(&mut self, ts: u64) {
+        self.hint_max_ts = Some(ts)
```

Comment:
Could you check if `hint_max_ts` < `hint_min_ts`?

Reply:
I would be better just keep it like lower_bound and upper_bound and if user setting
hint_max_ts < hint_hin_ts the iterator may returning no data ?